File: Backend/C#/SuperHero.ExternalService/Lookup/ExternalApiLookup.cs
using Microsoft.Extensions.Options;
using Newtonsoft.Json;
using SuperHero.Domain.Behavior.Repository;
using SuperHero.Domain.Model.CustomModel;
using SuperHero.ExternalService.Behavior;
using SuperHero.ExternalService.Settings;

namespace SuperHero.ExternalService.Lookup
{
    public class ExternalApiLookup : IExternalApiLookup
    {
        private readonly IRestService _restService;
        IOptions<HeroApiSettings> _heroApiSettings;

        public ExternalApiLookup(IRestService restService,
                                 IOptions<HeroApiSettings> heroApiSettings)
        {
            _restService = restService; 
            _heroApiSettings = heroApiSettings;
        }

        public async Task<CompleteHero> GetCompleteHeroById(int heroId)
        {
            var restResponse = await _restService.DoRestCall($"{_heroApiSettings.Value.ApiURL}{_heroApiSettings.Value.ApiToken}", heroId.ToString());

            var apiResponseModel = JsonConvert.DeserializeObject<CompleteHero>(restResponse);

            return apiResponseModel;
        }
    }
}

File: Backend/JAVA/application/src/main/java/com/superhero/repository/lookup/ExternalApiLookup.java
package com.superhero.repository.lookup;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.superhero.domain.behavior.repository.lookup.IExternalApiLookup;
import com.superhero.domain.behavior.service.external_service.IRestService;
import com.superhero.domain.model.custom_model.CompleteHero;

@Component
public class ExternalApiLookup implements IExternalApiLookup {

    @Autowired
    private final IRestService restService;

    @Value( "${heroapi.token}" )
    private String apiToken;
    
    @Value( "${heroapi.url}" )
    private String apiUrl;

    public ExternalApiLookup(IRestService restService) {
        this.restService = restService;
    }

    @Override
    public CompleteHero GetCompleteHeroById(int heroId) {
        // TODO Auto-generated method stub
        String apiResponse = this.restService.DoRestCall(String.format("%s%s", apiUrl, apiToken), String.valueOf(heroId));
        
        ObjectMapper mapper = new ObjectMapper();
        try {
            CompleteHero hero = mapper.readValue(apiResponse, CompleteHero.class);

            if(hero != null)
                return hero;

        } catch (JsonMappingException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        } catch (JsonProcessingException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        

        return new CompleteHero();
    }

}
